src/main/flight/imu.h

```diff
@@ -28,6 +28,11 @@ extern int accSumCount;
 extern float accVelScale;
 extern int32_t accSum[XYZ_AXIS_COUNT];
 
+typedef struct {
+    float w,x,y,z;
+    float ww,wx,wy,wz,xx,xy,xz,yy,yz,zz;
+} quaternion;
+#define DEFINE_QUATERNION(X) STATIC_UNIT_TESTED quaternion X = {.w=1, .x=0, .y=0,.z=0, .ww=1, .wx=0, .wy=0, .wz=0, .xx=0, .xy=0, .xz=0, .yy=0, .yz=0, .zz=0}
```

Comment:
`DEFINE_QUATERNION(q);` is opaque, there is no hint that quaternion is declared `STATIC_UNIT_TESTED` and initialized. Also `extern` declaration has to use `quaternion` explicitly, so almost nothing is abstracted with this macro (`DECLARE_QUATERNION` is one way to handle this, but `extern` will need some handling).
```
STATIC_UNIT_TESTED quaternion headfree = QUATERNION_ZERO_INITIALIZER;
```
is almost self-documenting (maybe QUATERNION_NULL_INITIALIZER is better name, quaternion is initialized to null rotation, not zeros). Also it is easy to define non-initialized quaternion consistently.

Using macro inside function is not about visibility, it is about `static`. In unittest, quaternion is initialized on each function call (or when scope is entered), but when building firmware, it is `static` - last value is reused. This will quickly lead to hard to find bugs ...